roles/setup_hammerdb/tasks/update_pg_hba.yml

```diff
@@ -0,0 +1,35 @@
+---
+
+- name: Reference pg_type variables
+  include_vars: "{{ role_path }}/../init_dbserver/vars/{{ pg_type }}.yml"
+
+- name: Initialize the hba_entries variable
+  set_fact:
+    hba_entries: >-
+      [
+        {
+          'users': 'all',
+          'databases': 'all',
+          'contype': 'host',
+          'source': '{{ hostvars[inventory_hostname].hammerdb_server_private_ip | default("127.0.0.1") }}/32',
+          'method': 'trust'
+        },
+        {
+          'users': 'all',
+          'databases': 'all',
+          'contype': 'host',
+          'source': '{{ hostvars['primary1'].private_ip | default("127.0.0.1") }}/32',
```

Comment:
For getting primary's private IP address, I would suggest to use the `pg_sr_cluster_nodes` lookup filter:
```yaml
- name: Get the primary information
  set_fact:
    primary_public_ip: "{{ node.ansible_host }}"
    primary_private_ip: "{{ node.private_ip }}"
  loop: "{{ lookup('edb_devops.edb_postgres.pg_sr_cluster_nodes', wantlist=True) }}"
  loop_control:
    loop_var: node
  when:  node.node_type == 'primary'
  no_log: "{{ disable_logging }}"
```

There is no guaranty that primary's inventory hostname is `primary1`.